### ISSUE TITLE
JsonInclude.Include.NON_NULL not honoured for Map

### ISSUE THREAD

**Issue by axelfontaine** [FasterXML/jackson-databind#1442]
Jackson Version: 2.8.4

```
Map<String, Object> in = new TreeMap<>();
in.put("abc", "123");
in.put("def", null);
in.put("ghi", 456);

String json = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL).writeValueAsString(in);

assertThat(json, not(containsString("def")));
assertThat(json, not(containsString("null")));
```

Even though the json string shouldn't contain "def" or "null", it still does.

**Comment by cowtowncoder** [comment]
Documentation problem; for structured types, basic inclusion refers to value itself, `Map`, which is not null. To exclude contents, you need to use new (in 2.7) method `setPropertyInclusion()` which allows specifying both "value inclusion" (Map) and "content inclusion" (elements). Something like:

    mapper.setPropertyInclusion(JsonInclude.Value.construct(Include.NON_NULL, Include.NON_NULL));

I hope this helps.

**Comment by cowtowncoder** [comment]
Created #1444, implemented, so that for 2.9 inclusion should work as you expected.
